Incorrect lower bound on http-client in http-conduit

https://hackage.haskell.org/package/http-client-0.6.2/docs/Network-HTTP-Client.html#v:shouldStripHeaderOnRedirect

```
Building library for http-conduit-2.3.6..
[1 of 3] Compiling Network.HTTP.Client.Conduit ( Network/HTTP/Client/Conduit.hs, dist/build/Network/HTTP/Client/Conduit.o )
[2 of 3] Compiling Network.HTTP.Conduit ( Network/HTTP/Conduit.hs, dist/build/Network/HTTP/Conduit.o )

Network/HTTP/Conduit.hs:161:7: error:
    Not in scope: ‘shouldStripHeaderOnRedirect’
    |
161 |     , shouldStripHeaderOnRedirect
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[3 of 3] Compiling Network.HTTP.Simple ( Network/HTTP/Simple.hs, dist/build/Network/HTTP/Simple.o )

Network/HTTP/Simple.hs:98:1: warning: [-Wunused-imports]
    The import of ‘Data.Monoid’ is redundant
      except perhaps to import instances from ‘Data.Monoid’
    To import instances alone, use: import Data.Monoid()
   |
98 | import Data.Monoid
   | ^^^^^^^^^^^^^^^^^^
```

Reason of this build plan is

```
[__3] rejecting: amazonka-core-1.6.1, amazonka-core-1.6.0 (conflict:
http-client==0.6.2, amazonka-core => http-client>=0.4 && <0.6)
```

--- a/http-conduit/Network/HTTP/Conduit.hs
+++ b/http-conduit/Network/HTTP/Conduit.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE FlexibleContexts  #-}
 {-# LANGUAGE OverloadedStrings #-}
 -- |
@@ -158,7 +159,9 @@ module Network.HTTP.Conduit
     , rawBody
     , decompress
     , redirectCount
+#if MIN_VERSION_http_client(0,6,2)
     , shouldStripHeaderOnRedirect
+#endif
     , checkResponse
     , responseTimeout
     , cookieJar

